docroot/profiles/custom/sitenow/src/ConfigOverride/GoogleAnalyticsOverride.php

```diff
@@ -11,18 +12,44 @@
  */
 class GoogleAnalyticsOverride implements ConfigFactoryOverrideInterface {
 
+  /**
+   * The config factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface|null
+   */
+  private $configFactory;
+
+  /**
+   * Constructs a new GoogleAnalyticsOverride object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface|null $config_factory
+   *   The config factory service.
+   */
+  public function __construct(?ConfigFactoryInterface $config_factory = NULL) {
+    $this->configFactory = $config_factory;
+  }
+
   /**
    * {@inheritdoc}
    */
   public function loadOverrides($names) {
     $overrides = [];
     if (in_array('google_analytics.settings', $names)) {
+      // Remove GA for local development.
       $env = getenv('AH_SITE_ENVIRONMENT');
-
-      // The GA module will choke on a NULL value.
       if ($env !== 'prod') {
         $overrides['google_analytics.settings']['account'] = '';
       }
+
+      if ($this->configFactory) {
```

Comment:
I think this->configFactory is always non-NULL based on you bringing it in via dependency injection. Instead it should check that the config for uiowa_core exists.

I think this could be simplified to:
```php
    $uiowa_core_config = $this->configFactory->get('uiowa_core.settings');
    if ($uiowa_core_config && !$uiowa_core_config->get('uiowa_core.ga')) {
      $overrides['google_analytics.settings']['account'] = '';
    }
```

Reply:
Care if I make this change @bspeare ?

Reply:
Please do! Thanks!

Reply:
Actually, I am getting TypeErrors elsewhere related to this. Poking.

Reply:
i think there is an issue with updating the services file like that (replacing request_stack). hard to tell if that will be an issue on deployment, but it appears to be an issue on syncing locally. We could add a `drupal_flush_caches` to the update hook but it may error before that.

```
TypeError: Drupal\sitenow\ConfigOverride\GoogleAnalyticsOve  
  rride::__construct(): Argument #1 ($config_factory) must be  
   of type Drupal\Core\Config\ConfigFactoryInterface, Symfony  
  \Component\HttpFoundation\RequestStack given, called in /va  
  r/www/html/docroot/core/lib/Drupal/Component/DependencyInje  
  ction/Container.php on line 261 in Drupal\sitenow\ConfigOve  
  rride\GoogleAnalyticsOverride->__construct() (line 28 of /v  
  ar/www/html/docroot/profiles/custom/sitenow/src/ConfigOverr  
  ide/GoogleAnalyticsOverride.php).  
```

Reply:
What site did you get that on?

Reply:
I had originally added the `if ($this->configFactory)` to get the test to pass that just failed on this build.

Reply:
Any site I previously had pulled down and did a `blt ds` on. (e.g. sandbox, admissions)

Reply:
steps to reproduce:

- checkout main
- sql-drop, cr, sql-sync
- uli. hit pages.
- checkout ga_delete
- hit pages (WSOD)
- updb
- hit pages (good?)